pinot-core/src/main/java/org/apache/pinot/core/operator/transform/transformer/datetime/DateTimeTransformerFactory.java

```diff
@@ -29,23 +30,39 @@ private DateTimeTransformerFactory() {
 
   public static BaseDateTimeTransformer getDateTimeTransformer(String inputFormatStr, String outputFormatStr,
       String outputGranularityStr) {
+    return getDateTimeTransformer(inputFormatStr, outputFormatStr, outputGranularityStr, null);
+  }
+
+  public static BaseDateTimeTransformer getDateTimeTransformer(String inputFormatStr, String outputFormatStr,
+      String outputGranularityStr, String bucketTimeZoneStr) {
+
     DateTimeFormatSpec inputFormat = new DateTimeFormatSpec(inputFormatStr);
     DateTimeFormatSpec outputFormat = new DateTimeFormatSpec(outputFormatStr);
     DateTimeGranularitySpec outputGranularity = new DateTimeGranularitySpec(outputGranularityStr);
+    DateTimeZone bucketingTz = null;
+    if (bucketTimeZoneStr != null) {
+      try {
+        // we're not using TimeZone.getTimeZone() because it's globally synchronized and returns default TZ when str
```

Comment:
Couldn't we use TimeZone.getTimeZone(ZoneId)?

Reply:
I guess we could, if ZoneId is obtained with .of() . I added comment to explain why I'm not using TimeZone.getTimeZone() like in e.g. DateTimeFormatPatternSpec and several other places .

Reply:
Replaced timezone parsing with ZoneId.of().